data-exporter/configureDataExporter.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+DIR=$(dirname "$(readlink -f "$0")") # Directory of the script -- allows the script to invoked from anywhere
+cd $DIR
+
+echo -n "Enter Neo4J user name: "
+read user
+if [ -z $user ]; then
+    echo "ERROR: Neo4J user required" >&2
+    exit 1;
+fi
+
+echo -n "Enter Neo4J password: "
+# Outputs asterisks instead of plain text
+# Taken from https://stackoverflow.com/a/4316755
+unset password;
+while IFS= read -r -s -n1 pass; do
+  if [[ -z $pass ]]; then
+     echo
+     break
```

Comment:
I don't really understand this condition. If they do not provide a password, then echo a blank line and break the iterative cycle? Why is there an iteration, and why echo nothing?

Reply:
I got this from an answer on Stack Overflow on how to show asterisks during a password prompt. 
 I don't know the logic inside out, but I believe it is:

* reading a single character at a time into $pass
* outputting an asterisk on screen for that character
* appending the actual character in $pass to $password
* when the user hits enter, doing an empty "echo" for outputting a newline and breaking out of the loop that reads one character per iteration

If they provide no password, then $password will also be empty.